src/layouts/MainLayout.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { GlobalHeader } from '@newrelic/gatsby-theme-newrelic';
+import { graphql, useStaticQuery } from 'gatsby';
+import { css } from '@emotion/core';
+
+const MainLayout = ({ children }) => {
+  const {
+    site: { layout },
+  } = useStaticQuery(graphql`
+    query {
+      site {
+        layout {
+          contentPadding
+          maxWidth
+        }
+      }
+    }
+  `);
+
+  return (
+    <>
+      <GlobalHeader />
+      <main
+        css={css`
+          margin: 0 auto;
```

Comment:
This comes from the theme correct? Or I guess, why is it that we query for this an add it to main? wouldn't make sense to add this some global styles that applies for every instance of the theme?

Reply:
> why is it that we query for this

Could you clarify what `this` is in this context? I'm not sure I follow your question.

All we are doing here is wrapping the content of the page in a `main` tag and applying these styles to `main`. The `margin: 0 auto` centers the element in the middle of the page. Without this, the layout would show up on the left side of the screen. We don't currently have anything in the theme that targets the `main` tag, so nothing here is coming from the theme.

This is not expected to be permanent. I wanted to docs to look just a little bit nicer than the super raw format (that previously had no spacing on the edge of the content, no global header, etc). Once we dive more into the navigation, I would expect this to change. Definitely take the implementation of this layout with a grain of salt since its not final or permanent 😄

Reply:
Sorry yes definitely should be more specific! What I meant was would it make more sense to add this as a global style? I guess my confusion runs from the fact that we have this `MainLayout` page that just adds the styles, which seem to be stitched into the theme. I guess my question is why don't we have a `MainLayout` component for the theme if it just queries margin and width that is available in the theme?

Reply:
Ah that is a great question! We've held off on moving layout specific stuff over to the theme because we wanted to be sure there was actually something to share between the 3 sites. For example, the developer site has a 2-column layout with a sidebar, where the open source site has a single-column layout with top-level navigation. There's not a lot to share there yet.

I'd rather have some duplication between the developer site and docs site for now to fully understand what makes sense to abstract and what not to. We may end up with a layout structure that doesn't fully match the developer site, in which case abstracting early might cause more headaches if we have to override a bunch of the work done in the theme. Duplication is far better than the wrong abstraction, and I want to wait to understand what the abstraction might look like before we try and pull it into the theme, especially since this implementation is not the final implementation and will likely change quite a bit between now and GA.